### ISSUE TITLE
An interesting part of soft-argmin

### ISSUE THREAD

**Issue by pobingwanghai** [JiaRenChang/PSMNet#114]
I notice the soft-argmin function use the -c for the softmax, while your realization and my experiments all showed that directly using c for the softmax can reach the required performance.
Do you have any idea of this observation?

**Comment by pobingwanghai** [comment]
This actually leads to a different explanation of the cost volume, where using -c means we deem the cost volume as the matching cost and using c means we deem the cost volume as the similarity.

**Comment by pobingwanghai** [comment]
I recheck the code of one of my previous paper, I use the -c in that paper to reproduce the GC-Net. It seems whether using -c or c do not greatly influence the performance. But this actually misguides how we explain this process and the cost volume. Computing the matching cost and similarity is not a totally dual problem, especially when we do not normalize the matching result.

**Comment by pobingwanghai** [comment]
Oh, I just find some differences between the two realizations. Using C leads to a more sharp result of softmax, where the maximum probability becomes more close to 1. This is sensible that the previous layers before the last 3d conv layer contributes to restrain the negative part. When we using -c form, the positive part becomes more compact than using the c form.

**Comment by pobingwanghai** [comment]
@JiaRenChang Would you like to explain the difference between the soft-argmin formulation in your paper and the real realization?

**Comment by JiaRenChang** [comment]
@pobingwanghai 
I agreed that using c or -c do not actually influence the performance. 
This achieved by the feature-based cost volume which concatenated left and right features to build cost volume. 
The network can learn similarity or matching cost depending on c or -c.
The formulation of soft-argmin is defined to learn matching cost in the paper.

**Comment by pobingwanghai** [comment]
@JiaRenChang Agreed. But, if we define the soft-argmin to handle the matching cost, then we must use the -c. It might be hard to change the definition of soft-argmin since the GC-Net. So it might be necessary to realize the soft-argmin with -c. Your PSMNet is such a great work and widely used as the baseline for many later works. Retraining the model with -c formulation is too inconvenient for this project. But I hope you can make it clear for the other researches to formulate the disparity computation process, maybe by an explanation in the code.

**Comment by JiaRenChang** [comment]
@pobingwanghai 
Sure, it's thankful for your advice. 
I will add comments in the code as soon as possible.
